File: android/lib/interfaces/counter-event-description.dart
import 'package:polydiff/interfaces/event-description.dart';
import 'package:polydiff/pages/replay-page.dart';

class CounterEventDescription extends EventDescription {
  final List<int> userCounter;

  const CounterEventDescription(super.time, this.userCounter);

  CounterEventDescription.fromJson(Map<String, dynamic> json)
      : userCounter = json['userCounter'].cast<int>(),
        super.fromJson(json);

  Map<String, dynamic> toJson() {
    return {
      'time': time,
      'userCounter': userCounter,
    };
  }

  @override
  play(ReplayPage replayPage) {
    print("CounterEventDescription.play() called.");
    replayPage.gameKey.currentState?.widget.createState().gameInfo.playerCounts[0] = userCounter[0];
    replayPage.gameKey.currentState?.widget.createState().gameInfo.playerCounts[1] = userCounter[1];
    replayPage.gameKey.currentState?.widget.createState().gameInfo.playerCounts[2] = userCounter[2];
    replayPage.gameKey.currentState?.widget.createState().gameInfo.playerCounts[3] = userCounter[3];
  }
}

File: client/src/app/classes/counter-event-description.ts
import { ReplayPageComponent } from "@app/pages/replay-page/replay-page.component";
import { EventDescription } from "./event-description";

export class CounterEventDescription extends EventDescription {
    private userCounter: number[];

    constructor(time: number, userCounter: number[]) {
        super(time);
        this.userCounter = userCounter;
    }

    play(replayPage: ReplayPageComponent) {
        replayPage.gamePageElem.counter.count = this.userCounter[0];
        replayPage.gamePageElem.counter2.count = this.userCounter[1];
        replayPage.gamePageElem.counter3.count = this.userCounter[2];   
        replayPage.gamePageElem.counter4.count = this.userCounter[3];

    }
}